File: scripts/maintenance/check_build_status.js
const Database = require('better-sqlite3');
const path = require('path');

const dbPath = path.join(__dirname, 'service', 'deadlock.sqlite3');
const db = new Database(dbPath, { readonly: true });

console.log('=== Build System Status After Fix ===\n');

// Check recent tasks
const recentTasks = db.prepare(`
  SELECT id, type, status, error,
         datetime(created_at, 'unixepoch') as created,
         datetime(finished_at, 'unixepoch') as finished
  FROM steam_tasks
  WHERE type = 'BUILD_PUBLISH'
  ORDER BY id DESC
  LIMIT 10
`).all();

console.log('Recent BUILD_PUBLISH Tasks:');
console.log('─'.repeat(80));
recentTasks.forEach(task => {
  const status = task.status === 'DONE' ? '✓' : task.status === 'FAILED' ? '✗' : '◷';
  console.log(`${status} Task #${task.id} - ${task.status}`);
  console.log(`  Created: ${task.created || 'N/A'}`);
  console.log(`  Finished: ${task.finished || 'N/A'}`);
  if (task.error) {
    console.log(`  Error: ${task.error.substring(0, 80)}`);
  }
  console.log('');
});

// Count by status
const statusCounts = db.prepare(`
  SELECT status, COUNT(*) as count
  FROM steam_tasks
  WHERE type = 'BUILD_PUBLISH'
  GROUP BY status
`).all();

console.log('\nTask Status Summary:');
console.log('─'.repeat(80));
statusCounts.forEach(row => {
  console.log(`  ${row.status}: ${row.count}`);
});

// Check clone status
const cloneCounts = db.prepare(`
  SELECT status, COUNT(*) as count
  FROM hero_build_clones
  GROUP BY status
`).all();

console.log('\nBuild Clone Status Summary:');
console.log('─'.repeat(80));
cloneCounts.forEach(row => {
  console.log(`  ${row.status}: ${row.count}`);
});

// Check most recent successful upload
const recentSuccess = db.prepare(`
  SELECT c.origin_hero_build_id, c.uploaded_build_id, c.uploaded_version,
         datetime(c.updated_at, 'unixepoch') as updated,
         s.hero_id, s.name
  FROM hero_build_clones c
  JOIN hero_build_sources s ON c.origin_hero_build_id = s.hero_build_id
  WHERE c.status = 'uploaded'
  ORDER BY c.updated_at DESC
  LIMIT 5
`).all();

if (recentSuccess.length > 0) {
  console.log('\nRecent Successful Uploads:');
  console.log('─'.repeat(80));
  recentSuccess.forEach(row => {
    console.log(`✓ Build #${row.origin_hero_build_id} → #${row.uploaded_build_id} v${row.uploaded_version}`);
    console.log(`  Hero: ${row.hero_id} | Name: ${row.name}`);
    console.log(`  Uploaded: ${row.updated}`);
    console.log('');
  });
} else {
  console.log('\n⚠ No successful uploads yet');
}

// Check Steam bridge status
const steamStatus = db.prepare(`
  SELECT payload,
         datetime(last_heartbeat, 'unixepoch') as last_heartbeat
  FROM standalone_bot_state
  WHERE bot = 'steam'
  LIMIT 1
`).get();

if (steamStatus) {
  const payload = JSON.parse(steamStatus.payload || '{}');
  const runtime = payload.runtime || {};

  console.log('\nSteam Bridge Status:');
  console.log('─'.repeat(80));
  console.log(`  Last heartbeat: ${steamStatus.last_heartbeat}`);
  console.log(`  Logged in: ${runtime.logged_on ? '✓' : '✗'}`);
  console.log(`  GC Ready: ${runtime.deadlock_gc_ready ? '✓' : '✗'}`);
}

db.close();

File: check_build_status.py
import sqlite3
import json
from datetime import datetime, timedelta

DB_PATH = r"C:\Users\Nani-Admin\Documents\Deadlock\service\deadlock.sqlite3"

conn = sqlite3.connect(DB_PATH)
c = conn.cursor()

print("=" * 60)
print("BUILD SYSTEM STATUS CHECK")
print("=" * 60)

# Check build sources
print("\n=== Build Sources ===")
total_sources = c.execute("SELECT COUNT(*) FROM hero_build_sources").fetchone()[0]
print(f"Total build sources: {total_sources}")

# Get most recent source
recent = c.execute("""
    SELECT name, fetched_at
    FROM hero_build_sources
    ORDER BY fetched_at DESC
    LIMIT 1
""").fetchone()
if recent:
    print(f"Most recent: {recent[0]} at {datetime.fromtimestamp(recent[1])}")

# Check clone queue status
print("\n=== Clone Queue Status ===")
for row in c.execute("SELECT status, COUNT(*) FROM hero_build_clones GROUP BY status"):
    print(f"  {row[0]}: {row[1]}")

# Get recent activity
print("\n=== Recent Clone Activity (last 24h) ===")
cutoff = int((datetime.now() - timedelta(hours=24)).timestamp())
recent_clones = c.execute("""
    SELECT status, COUNT(*)
    FROM hero_build_clones
    WHERE created_at > ?
    GROUP BY status
""", (cutoff,)).fetchall()
if recent_clones:
    for row in recent_clones:
        print(f"  {row[0]}: {row[1]}")
else:
    print("  No activity in last 24h")

# Check pending builds
pending_count = c.execute("SELECT COUNT(*) FROM hero_build_clones WHERE status = 'pending'").fetchone()[0]
print(f"\n=== Pending Builds ===")
print(f"Total pending: {pending_count}")

if pending_count > 0:
    print("\nSample pending builds:")
    for row in c.execute("""
        SELECT hbs.name, hbc.created_at
        FROM hero_build_clones hbc
        JOIN hero_build_sources hbs ON hbc.origin_hero_build_id = hbs.hero_build_id
        WHERE hbc.status = 'pending'
        ORDER BY hbc.created_at DESC
        LIMIT 5
    """):
        print(f"  - {row[0]} (created {datetime.fromtimestamp(row[1])})")

# Check Steam Bridge status
print("\n=== Steam Bridge Status ===")
steam_row = c.execute("SELECT payload, updated_at FROM standalone_bot_state WHERE bot = 'steam'").fetchone()
if steam_row:
    try:
        payload = json.loads(steam_row[0])
        print(f"Last update: {steam_row[1]}")
        print(f"deadlock_gc_ready: {payload.get('deadlock_gc_ready', False)}")
        print(f"steam_connected: {payload.get('steam_connected', False)}")
        print(f"gc_connected: {payload.get('gc_connected', False)}")
    except:
        print("Error parsing Steam Bridge payload")
else:
    print("Steam Bridge not found in DB")

# Check steam tasks
print("\n=== Steam Tasks ===")
task_stats = c.execute("""
    SELECT type, status, COUNT(*)
    FROM steam_tasks
    WHERE type = 'BUILD_PUBLISH'
    GROUP BY type, status
""").fetchall()
if task_stats:
    for row in task_stats:
        print(f"  {row[0]} - {row[1]}: {row[2]}")
else:
    print("  No BUILD_PUBLISH tasks found")

# Recent tasks
print("\n=== Recent Steam Tasks (last 24h) ===")
recent_tasks = c.execute("""
    SELECT status, COUNT(*)
    FROM steam_tasks
    WHERE created_at > ?
    AND type = 'BUILD_PUBLISH'
    GROUP BY status
""", (cutoff,)).fetchall()
if recent_tasks:
    for row in recent_tasks:
        print(f"  {row[0]}: {row[1]}")
else:
    print("  No tasks in last 24h")

conn.close()
print("\n" + "=" * 60)
